### ISSUE TITLE
fake camera: specify IP address to bind rather than network interface name

### ISSUE THREAD

**Issue by eudoxos** [AravisProject/aravis#535]
**Is your feature request related to a problem? Please describe.**
The current fake camera has the `interface-name` property, the default `lo` is not portable. Camera setup cycles through interfaces only to obtain the right IP address to bind.

**Describe the solution you'd like**
Instead of specifying `interface-name`, IP address (or hostname) should be specified. The default could be `127.0.0.1`. The loop over interfaces could be also removed. Most UNIX servers specify IP adress they bind, so it would be culturally consistent.

**Describe alternatives you've considered**
Finding loopback interface name under windows (and use that as the default) -- until I found that loopback (mostly) does not even exist as network interface under Windows.

Another alternative is to have the option to specify either the interface or the IP address. This would probably be superfluous and add unnecessary complexity for little use.

**Additional context**
Win32 port is the incentive to bring this up.

**Comment by eudoxos** [comment]
I missed your new branch and did something similar in #538. I think the new functions in arvnetwork (`arv_network_get_interface_by_name` and `arv_network_get_fake_ipv4_loopback`) could still be useful.

The easiest solution would be (probably) to bind the fake camera to all interfaces by default (`INADDR_ANY` which is `0.0.0.0`)  but I am not sure how that plays with camera discovery; probably quite bad :skull: 

What was not clear to me (when I was thinking about it) was how to get the correct netmask/broadcast knowing the IP address only...

**Comment by eudoxos** [comment]
I'd suggest the fake camera to go like this (with `name` being the address-as-string / interface name):

1. try `arv_network_get_interface_by_address(name)`, use that if something found;
2. otherwise try `arv_network_get_interface_by_name(name)`, use that if something found;
3. otherwise use `arv_network_get_fake_ipv4_loopback`.

As an alternative, the fake ipv4 loopback could be added to the result of `arv_enumerate_network_interfaces` under windows automatically, this would make the 3rd point unnecessary.

**Comment by eudoxos** [comment]
I took it a step further to iron out other windows issues of the fake camera (`255.255.255.255` does not bind at all, loopback broadcast seems to fail as well), plus I incorporated your idea about specifying address or interface name. I am sorry I did not notice the work you've done already.

The CI curiously failed once in evaluator test on i686, but not another time the exact source code was built... strange.

![image](https://user-images.githubusercontent.com/1029876/126867155-b8f54fc8-98b5-4621-8351-5701d39ea2a7.png)
